exonum/src/blockchain/transaction.rs

```diff
@@ -353,31 +353,37 @@ pub trait TransactionSet
 /// `transactions!` is used to declare a set of transactions of a particular service.
 ///
 /// The macro generates a type for each transaction and a helper enum which can hold
-/// any of the transactions. You must implement `Transaction` trait for each of the
+/// any of the transactions. You need to implement the `Transaction` trait for each of the
 /// transactions yourself.
 ///
-/// See `Service` trait documentation for a full example of usage.
+/// See [`Service`] trait documentation for a full example of usage.
 ///
-/// Each transaction is specified as a Rust struct. For additional reference about
-/// data layout see the documentation of the [`encoding` module](./encoding/index.html).
+/// Each transaction is specified as a Rust struct. For additional information about
+/// data layout see the documentation on the [`encoding` module](./encoding/index.html).
 ///
-/// Additionally, the macro must define identifier of a service, which will be used
+/// Additionally, the macro must define the identifier of a service, which will be used
 /// [in parsing messages][parsing], as `const SERVICE_ID`. Service ID should be unique
 /// within the Exonum blockchain.
 ///
-/// For each transaction the macro creates getter methods for all fields with the same names as
-/// fields. In addition, two constructors are defined:
+/// For each transaction the macro creates getter methods for all defined fields.
+/// The names of the methods coincide with the field names. In addition,
+/// two constructors are defined:
 ///
-/// - `new` takes all fields in the order of their declaration in the macro, and a [`SecretKey`]
-///   to sign the message as the last argument.
-/// - `new_with_signature` takes all fields in the order of their declaration in the macro,
-///   and a message [`Signature`].
+/// - `new` accepts as arguments all fields in the order of their declaration in
+/// the macro, and a [`SecretKey`] as the last argument. A `SecretKey` is used
```

Comment:
Nitpick: I suggest to preserve indentation in list formatting:

```markdown
- Lorem ipsum dolor sit amet, consectetur
  adipiscing elit.
- Suspendisse id arcu et quam hendrerit
  scelerisque.
```

This makes lists more readable and readily identifiable.